runtime/lua/vim/lsp/util.lua

```diff
@@ -1143,28 +1152,31 @@ function M.fancy_floating_markdown(contents, opts)
   api.nvim_win_set_option(winnr, 'conceallevel', 2)
   api.nvim_win_set_option(winnr, 'concealcursor', 'n')
 
-  vim.cmd("ownsyntax lsp_markdown")
   local idx = 1
   --@private
   local function apply_syntax_to_region(ft, start, finish)
-    if ft == '' then return end
+    if ft == '' then ft = "lsp_markdown" end
```

Comment:
Was able to fix it. Previously, the whole buffer was `markdown`, so this line skipped syn regions for code blocks without a language.

But now we need to markup those regions with markdown instead.

Reply:
Actually now that I think about it. Is it a good idea to use `lsp_markdown` for a code block without a lang?

For the `<pre>` blocks I actually used `plaintex` instead.

What do you think?

Reply:
I also wonder why the special handling is uberhaut needed. Shouldn't `vim_markdown_fenced_languages` be used instead and leave the code blocks as is?

Reply:
The main point of lsp_markdown is to apply conceal rules not offered by markdown to hide escaped characters. In the case of there not being a language tag, there shouldn't be a difference between markdown and plaintex. If there isn't a tag probably plaintex is fine, I doubt there's a noticeable perf difference.

Reply:
What special handling? The stripping of the code fences? It's mostly for aesthetics to avoid a bunch of concealed blank lines IIRC

Reply:
never mind. It's prettier when using the regions :)

Reply:
As an alternative, we can leave those code blocks ` ``` ` without lang and let the markdown render them as a code block:
![image](https://user-images.githubusercontent.com/292349/119053506-00c5a180-b97b-11eb-94e1-3e6cd90a1651.png)

Reply:
@mjlbach so should I change it to use `plaintex` instead? Then of course those darker `>>>` lines would be normal text. Or should I use a syn match and apply the markdown code block highlight group instead of a region?

Reply:
I don't think either is more correct, you've proven to have an aesthetic eye for these things :) I learn towards *not* using plaintex but if someone complains we can always change it later

Reply:
Maybe something like this instead?

![image](https://user-images.githubusercontent.com/292349/119055399-d45f5480-b97d-11eb-925f-84d4047cfe17.png)

Reply:
Should be `markdownCodeBlock` but the idea is the same :)

I'm happy to keep it as was before and use `lsp_markdown` for code blocks without a language too. Let me know what you prefer.

Reply:
Would look like this then:
![image](https://user-images.githubusercontent.com/292349/119056875-49cc2480-b980-11eb-8277-65e6d2c01fee.png)

Reply:
That looks good IMO, do you agree?

Reply:
Yep, I prefer it that way. Also makes the most sense I guess. When a code block has a `lang`, we render it using that syntax, otherwise we render it as a code block.

I just pushed an update.

The `<pre>` is also rendered as a code block now.

Reply:
I also fixed an issue I introduced with the seperators. The previous code expected a sep to be added for every block and shifted the highlight start and end. Since we now sometimes skip a sep, I had to change it to reflect this.

Reply:
Im going to get some sleep now, but do let me know if you want any additional changes. I can work on them tomorrow morning. Thanks.

another example:
![image](https://user-images.githubusercontent.com/292349/119057703-d0353600-b981-11eb-8295-5c2e239ac2b0.png)